ftplugin/markdown.vim

```diff
@@ -38,6 +38,18 @@ function! MarkdownFold()
     return ">2"
   endif
 
+  " Front matter
+  if v:lnum == 1 && line == '---'
+    let b:markdown_frontmatter = 1
+    return ">1"
+  endif
+
+  " End of front matter
+  if (line == '...') && b:markdown_frontmatter
```

Comment:
Isn't this supposed to end with `---` ?

Reply:
This will cause an error if `b:markdown_frontmatter` isn't defined.